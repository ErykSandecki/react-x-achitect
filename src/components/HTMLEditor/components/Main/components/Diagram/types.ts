// types
import { BlockType, PortType, T2DCoordinates, TBlock, TConnector } from 'types';

export type TDraggableConnectorSource = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnectorTarget = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnector = {
  showIndicator: boolean;
  source?: TDraggableConnectorSource;
  sourceCoordinates: T2DCoordinates;
  target?: TDraggableConnectorTarget;
  targetCoordinates: T2DCoordinates;
};

export type TPossiblePort = T2DCoordinates & {
  blockId: string;
  blockType: BlockType;
  height: number;
  portType: PortType;
  width: number;
};

export type TShadowConnector = Pick<
  TConnector,
  'joinOperator' | 'relatedWithBlockId' | 'towardsTo'
> & {
  sourceBlock: Partial<TBlock> & { portType: PortType };
  targetBlock: Partial<TBlock> & { portType: PortType };
};
