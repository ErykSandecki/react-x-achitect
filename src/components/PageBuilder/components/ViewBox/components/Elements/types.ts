// types
import { MouseMode } from 'components/PageBuilder/enums';
import { TElement } from 'types';

export type TElementProps = {
  className: string;
  id: TElement['id'];
  mouseMode: MouseMode;
  parentId: TElement['parentId'];
  type: TElement['type'];
};
