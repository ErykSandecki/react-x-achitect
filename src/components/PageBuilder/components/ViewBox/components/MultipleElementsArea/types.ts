// types
import { TElement, TRectCoordinates } from 'types';

export type TElements = {
  elementsCordinates: Array<{
    coordinates: TRectCoordinates;
    id: TElement['id'];
  }>;
  outline: TRectCoordinates;
};
