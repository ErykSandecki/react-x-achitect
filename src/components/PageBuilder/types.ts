// types
import { TRectCoordinates } from 'types';

export type TRectArea = TRectCoordinates | null;
