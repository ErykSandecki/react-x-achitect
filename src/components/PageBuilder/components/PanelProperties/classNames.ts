export const className = 'PanelProperties';

export const classNames = {
  [className]: className,
  areaHandleResize: `${className}__area-handle-resize`,
};
