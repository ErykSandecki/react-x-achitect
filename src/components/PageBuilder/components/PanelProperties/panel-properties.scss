.PanelProperties {
  &__area-handle-resize {
    cursor: ew-resize;
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 5px;
  }
}
