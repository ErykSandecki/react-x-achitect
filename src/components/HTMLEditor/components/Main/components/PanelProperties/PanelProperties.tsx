import { forwardRef, memo } from 'react';

// hooks
import { useTheme } from 'hooks/useTheme/useTheme';

// others
import { className, classNames } from './classNames';
import { PANEL_PROPERTIES_ID, PANEL_PROPERTIES_WIDTH } from './constants';

// styles
import styles from './panel-properties.scss';

const PanelProperties = forwardRef<HTMLDivElement, {}>((_, ref) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <div
      className={cx(classNamesWithTheme[className])}
      id={PANEL_PROPERTIES_ID}
      ref={ref}
      style={{ width: PANEL_PROPERTIES_WIDTH }}
    ></div>
  );
});

export default memo(PanelProperties);
