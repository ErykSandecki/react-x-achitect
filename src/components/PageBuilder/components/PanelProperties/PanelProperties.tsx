import { FC, memo, useRef } from 'react';

// components
import { Box, E2EDataAttribute } from 'shared';

// hooks
import { useResizeHandler, useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';
import { TOOLBAR_HEIGHT } from 'components/PageBuilder/constants';

// styles
import styles from './panel-properties.scss';

// types
import { E2EAttribute, ZIndex } from 'types';

export type TPanelPropertiesProps = {};

const PanelProperties: FC<TPanelPropertiesProps> = () => {
  const boxRef = useRef(null);
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { onMouseDownX, width } = useResizeHandler(
    0,
    240,
    window.innerHeight - TOOLBAR_HEIGHT,
    500,
    0,
    240,
    boxRef,
  );

  return (
    <Box
      classes={{ className: cx(classNamesWithTheme[className]) }}
      onMouseDown={(event) => event.stopPropagation()}
      ref={boxRef}
      style={{ touchAction: 'manipulation' }}
      sx={{
        bg: 'neutral5',
        borderLeft: 1,
        boxSizing: 'border-box',
        height: `calc(100vh - ${TOOLBAR_HEIGHT}px)`,
        position: 'absolute',
        right: 0,
        top: `${TOOLBAR_HEIGHT}px`,
        width: `${width}px`,
        zIndex: ZIndex.standard,
      }}
    >
      <E2EDataAttribute type={E2EAttribute.resize}>
        <div
          className={cx(classNamesWithTheme.areaHandleResize)}
          onMouseDown={(event) => onMouseDownX(event, true)}
        />
      </E2EDataAttribute>
    </Box>
  );
};

export default memo(PanelProperties);
