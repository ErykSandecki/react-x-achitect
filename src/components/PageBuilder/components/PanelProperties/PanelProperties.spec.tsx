import { fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';

// components
import PanelProperties from './PanelProperties';

// mocks
import 'test/mocks/sagas/allSagas';

// store
import { configureStore } from 'store/store';

// types
import { E2EAttribute, MouseButton } from 'types';

// utils
import { getByE2EAttribute } from 'test';

describe('PanelProperties snapshots', () => {
  it('should render PanelProperties', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <PanelProperties />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with changed width', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment, container } = render(
      <Provider store={store}>
        <PanelProperties />
      </Provider>,
    );

    // action
    fireEvent.mouseDown(getByE2EAttribute(container, E2EAttribute.resize), {
      buttons: MouseButton.lmb,
    });
    fireEvent.mouseMove(document, { clientX: 500 });

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
