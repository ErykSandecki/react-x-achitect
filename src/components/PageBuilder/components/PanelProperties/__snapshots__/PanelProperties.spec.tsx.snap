// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelProperties snapshots should render PanelProperties 1`] = `
<DocumentFragment>
  <div
    class="css-i7rldk PanelProperties PanelProperties--light"
    data-test-box="panel-properties"
  >
    <div
      class="PanelProperties__area-handle-resize PanelProperties__area-handle-resize--light"
      data-test-resize=""
    />
    <div
      class="css-q0izro Section Section--light"
      data-test-section=""
    >
      <small
        class="css-q0izro Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Mode
      </small>
      <div
        class="css-q0izro Tabs Tabs--light"
        data-test-box=""
      >
        <div
          class="Tabs__tab Tabs__tab--light Tabs__tab--active Tabs__tab--active--light"
          data-test-tab="design"
        >
          Design
        </div>
        <div
          class="Tabs__tab Tabs__tab--light"
          data-test-tab="prototype\`,"
        >
          Prototype
        </div>
      </div>
    </div>
    <div
      class="css-q0izro Section Section--light"
      data-test-section=""
    >
      <small
        class="css-q0izro Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Page
      </small>
      <div
        class="css-q0izro SectionColumn SectionColumn--light"
        data-test-section=""
      >
        <div
          class="SectionColumn__content SectionColumn__content--light"
        >
          <div
            class="css-q0izro FieldGroup FieldGroup--light"
            data-test-field-group=""
          >
            <div
              class="css-q0izro ColorPicker__hex-input ColorPicker__hex-input--light TextField TextField--light"
              data-test-text-field="color"
            >
              <div
                aria-describedby="test-id"
              >
                <div
                  class="css-q0izro Color Color--light"
                  data-test-color=""
                >
                  <div
                    class="Color__picker Color__picker--light"
                    style="background-color: rgb(241, 242, 246);"
                  />
                  <div
                    class="Color__picker-alpha Color__picker-alpha--light"
                    style="background-color: rgb(241, 242, 246);"
                  />
                  <div
                    class="Color__picker-texture Color__picker-texture--light"
                  />
                </div>
              </div>
              <input
                class="TextField__input TextField__input--light"
                data-test-text-field-input="color"
                maxlength="6"
                value="f1f2f6"
              />
            </div>
            <div
              class="css-q0izro ColorPicker__alpha-input ColorPicker__alpha-input--light TextField TextField--light"
              data-test-text-field="alpha"
            >
              <input
                class="TextField__input TextField__input--light"
                data-test-text-field-input="alpha"
                max="100"
                maxlength="6"
                min="0"
                type="number"
                value="100"
              />
              <div
                class="ScrubbableInput ScrubbableInput--light"
                data-test-scrubbable-input=""
              >
                %
              </div>
            </div>
          </div>
        </div>
        <div
          class="SectionColumn__buttons SectionColumn__buttons--light"
        >
          <div
            class="css-q0izro ButtonIcon ButtonIcon--light"
            data-test-button-icon=""
          >
            <div
              class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="EyesOpened"
              height="14"
              name="EyesOpened"
              width="14"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PanelProperties snapshots should render when some element is selected 1`] = `
<DocumentFragment>
  <div
    class="css-i7rldk PanelProperties PanelProperties--light"
    data-test-box="panel-properties"
  >
    <div
      class="PanelProperties__area-handle-resize PanelProperties__area-handle-resize--light"
      data-test-resize=""
    />
    <div
      class="css-q0izro Section Section--light"
      data-test-section=""
    >
      <small
        class="css-q0izro Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Mode
      </small>
      <div
        class="css-q0izro Tabs Tabs--light"
        data-test-box=""
      >
        <div
          class="Tabs__tab Tabs__tab--light Tabs__tab--active Tabs__tab--active--light"
          data-test-tab="design"
        >
          Design
        </div>
        <div
          class="Tabs__tab Tabs__tab--light"
          data-test-tab="prototype\`,"
        >
          Prototype
        </div>
      </div>
    </div>
    <div
      class="css-q0izro Section Section--light"
      data-test-section=""
    >
      <div
        class="css-q0izro SectionColumn SectionColumn--light"
        data-test-section=""
      >
        <div
          class="SectionColumn__content SectionColumn__content--light"
        >
          <small
            class="css-q0izro Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
            data-test-text=""
            style="text-align: inherit;"
          >
            Frame
          </small>
        </div>
        <div
          class="SectionColumn__buttons SectionColumn__buttons--light"
        >
          <div
            class="css-q0izro ButtonIcon ButtonIcon--light"
            data-test-button-icon=""
          >
            <div
              class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="HtmlTag"
              height="14"
              name="HtmlTag"
              width="14"
            />
          </div>
          <div
            class="css-q0izro ButtonIcon ButtonIcon--light"
            data-test-button-icon=""
          >
            <div
              class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="Component"
              height="14"
              name="Component"
              width="14"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PanelProperties snapshots should render with changed width 1`] = `
<DocumentFragment>
  <div
    class="css-18uqunt PanelProperties PanelProperties--light"
    data-test-box="panel-properties"
  >
    <div
      class="PanelProperties__area-handle-resize PanelProperties__area-handle-resize--light"
      data-test-resize=""
    />
    <div
      class="css-q0izro Section Section--light"
      data-test-section=""
    >
      <small
        class="css-q0izro Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Mode
      </small>
      <div
        class="css-q0izro Tabs Tabs--light"
        data-test-box=""
      >
        <div
          class="Tabs__tab Tabs__tab--light Tabs__tab--active Tabs__tab--active--light"
          data-test-tab="design"
        >
          Design
        </div>
        <div
          class="Tabs__tab Tabs__tab--light"
          data-test-tab="prototype\`,"
        >
          Prototype
        </div>
      </div>
    </div>
    <div
      class="css-q0izro Section Section--light"
      data-test-section=""
    >
      <small
        class="css-q0izro Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        Page
      </small>
      <div
        class="css-q0izro SectionColumn SectionColumn--light"
        data-test-section=""
      >
        <div
          class="SectionColumn__content SectionColumn__content--light"
        >
          <div
            class="css-q0izro FieldGroup FieldGroup--light"
            data-test-field-group=""
          >
            <div
              class="css-q0izro ColorPicker__hex-input ColorPicker__hex-input--light TextField TextField--light"
              data-test-text-field="color"
            >
              <div
                aria-describedby="test-id"
              >
                <div
                  class="css-q0izro Color Color--light"
                  data-test-color=""
                >
                  <div
                    class="Color__picker Color__picker--light"
                    style="background-color: rgb(241, 242, 246);"
                  />
                  <div
                    class="Color__picker-alpha Color__picker-alpha--light"
                    style="background-color: rgb(241, 242, 246);"
                  />
                  <div
                    class="Color__picker-texture Color__picker-texture--light"
                  />
                </div>
              </div>
              <input
                class="TextField__input TextField__input--light"
                data-test-text-field-input="color"
                maxlength="6"
                value="f1f2f6"
              />
            </div>
            <div
              class="css-q0izro ColorPicker__alpha-input ColorPicker__alpha-input--light TextField TextField--light"
              data-test-text-field="alpha"
            >
              <input
                class="TextField__input TextField__input--light"
                data-test-text-field-input="alpha"
                max="100"
                maxlength="6"
                min="0"
                type="number"
                value="100"
              />
              <div
                class="ScrubbableInput ScrubbableInput--light"
                data-test-scrubbable-input=""
              >
                %
              </div>
            </div>
          </div>
        </div>
        <div
          class="SectionColumn__buttons SectionColumn__buttons--light"
        >
          <div
            class="css-q0izro ButtonIcon ButtonIcon--light"
            data-test-button-icon=""
          >
            <div
              class="css-q0izro Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="EyesOpened"
              height="14"
              name="EyesOpened"
              width="14"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
